runners/direct-java/src/main/java/org/apache/beam/runners/direct/BundleExecutor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.direct;
+
+import org.apache.beam.runners.local.Bundle;
+
+/**
+ * An executor that is capable of processing some bundle of input over some executable stage or
+ * step.
+ */
+interface BundleExecutor<BundleT extends Bundle<?>, ExecutableT> {
```

Comment:
naming: `BundleProcessor`? A bundle is usually conceptualized as data, not so much executable, even though it _is_ tied to a step so it is unambiguous how it has to be processed.

Reply:
I like `processor`. Done.